azext_edge/edge/providers/check/base/pod.py

```diff
@@ -121,40 +114,91 @@ def _decorate_pod_condition(condition: bool) -> Tuple[str, str]:
 
         # When pod in obnormal state, sometimes the conditions are not available
         if pod_conditions:
+            conditions_readiness = True
+            conditions_display_list: List[Tuple[str, str]] = []
+            unknown_conditions_display_list: List[Tuple[str, str]] = []
+
+            for condition in pod_conditions:
+                type = condition["type"]
+
+                try:
+                    condition_type = POD_CONDITION_TEXT_MAP[type]
+                    condition_status = condition.get("status") == "True"
+                    conditions_readiness = conditions_readiness and condition_status
+                    pod_condition_deco, status = _decorate_pod_condition(condition=condition_status)
+                except KeyError:
+                    condition_type = type
+                    condition_status = condition.get("status")
+
+                formatted_reason = ""
+                condition_reason = condition.get("reason", "")
+
+                if condition_reason:
+                    # remove the [ and ] to prevent console not printing the text
+                    condition_reason = condition_reason.replace("[", "\\[")
```

Comment:
did you get the ] here too? or does is the rich linter smart enough to skip ]'s if it cannot find an opening [?